extensions/oidc-token-propagation-reactive/runtime/src/main/java/io/quarkus/oidc/token/propagation/reactive/AccessTokenRequestReactiveFilter.java

```diff
@@ -124,7 +124,9 @@ protected String getClientName() {
 
     public void propagateToken(ResteasyReactiveClientRequestContext requestContext, String accessToken) {
         if (accessToken != null) {
-            requestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, BEARER_SCHEME_WITH_SPACE + accessToken);
+            if (requestContext.getHeaders().get(HttpHeaders.AUTHORIZATION).stream().noneMatch(h -> h.toString().startsWith(BEARER_SCHEME_WITH_SPACE))) {
+                requestContext.getHeaders().add(HttpHeaders.AUTHORIZATION, BEARER_SCHEME_WITH_SPACE + accessToken);
```

Comment:
@sahuefficy Thanks, can it be simplified so that the header is not added but set ? I was expecting only change from `.add` to `.set` (or `put`). Your fix is correct but simply setting the header would be as cheap or even faster when the duplication situation is possible

Reply:
The problem with set/put is that it would override a non-bearer scheme authorization header. Thats why add is the right thing to do. I have indeed pushed another commit that actually checks for null before streaming.

Reply:
I have got your point and have changed the check as well as add to putSingle.

Reply:
Thanks @sahuefficy, but I think we should also drop the null check, as it leaves the same ambiguity open - if the header is already present  - may be it is the basic scheme - now we need to decide if we want to report an error or not - but it is really not a decision that should be taken at this filter's level. I'm sorry this simple PR which is technically correct is still being reviewed :-), I do appreciate your time and patience. Please also squash the commits, thanks

Reply:
Now the PR has become pretty simple with the add replaced with putSingle.